doc/relevancy.md

```diff
@@ -0,0 +1,442 @@
+
+The stopwords themselves seem to be served by elasticsearch. There's nothing on
+the Search API repo that looks like a list of omitted terms.
+
+There has been some talk of not including stopwords in the past
+https://github.com/alphagov/search-api/commit/0fe6e526c78e8b115371855a91d4b39ccb22098a
+during this time a no_stop method was implemented that seems to stop stopwords
+from being used. Again this seems to be served by ES because it's not defined
+on rummager.
+
+Given that we're taking a stock list of stopwords from ES, we may be missing
+out on a chance to edit them in a way that might be more useful for users.
+I'm not sure how we'd go about changing this but i'd guess that we'd need to
+look into what words could be useful to omit / not omit.
+
+If you search Search API for stopwords then you get literally nothing
+http://search-api.dev.gov.uk/search.json?q=the+that+and+if
+
+If you search for nothing then you get the some of the more frequently clicked links
+http://search-api.dev.gov.uk/search.json?q=
+I'm not sure how useful this is but I thought it was interesting
+
+## Synonyms
+
+A synonym is a word or phrase that means exactly or nearly the same as another
+word or phrase in the same language, for example shut is a synonym of close.
+
+Synonyms are defined in the [synonyms.yml file](https://github.com/alphagov/rummager/blob/master/config/schema/synonyms.yml).
+
+Each synonym group is represented as comma separated lists of synonyms, and
+optionally a "=>" symbol.
+
+### Grouping of synonyms
+
+The use of => is known as _Simple contraction_. It maps a group of synonyms on
+the left side to a single value on the right side:
+
+"leap,hop => jump"
+
+means "a search for 'leap' or 'hop' should return documents with 'jump' in them.
+
+It must be applied both at index time and at query time, to ensure that query
+terms are mapped to the same single value that exists in the index.
+
+Comma separated "foo, bar, baz" means to return any of those terms.
+
+### Filtering
+
+Additional configuration is defined in the elasticsearch_schema.yml and
+stems.yml files.  This configuration is merged with the JSON configuration,
+and then passed to elasticsearch directly.
+
+Within elasticsearch_schema.yml, as well as an index_synonym filter there is
+also a synonym_protwords filter.
+
+This marks the terms generated from the synonym matching process as "keywords".
+This prevents them being stemmed, and is required if there is a stemming filter
+following the synonym filter in the analysis chain. This will make sure phrases
+stay together aka "heart attack".
+
+This blog post suggests that as well as using keywords, it can be useful to use
+'keep words' to only filter the phrases we want.
+
+The post suggests to first generate shingles, followed by synonyms, followed
+by keepwords.
+
+In other words: generate candidate keyphrases by shingling, expand them with
+synonyms, then cull out any non-synonyms with keepwords.
+
+### Old synonyms
```

Comment:
I don't think the old synonyms are used.  The configuration is loaded: https://github.com/alphagov/search-api/blob/b6e0400c7e2397a927d702e24302b55bb58722d3/lib/schema/schema_config.rb#L49-L62

But nothing actually uses it.  There's nothing in the queries or analysers about the old synonyms.